### ISSUE TITLE
Ability to control other aria2c instance

### ISSUE THREAD

**Issue by bahusoid** [aria2/aria2#1524]
So it's a feature request to support aria2c to work as a proxy and send all commands via RPC to remote aria2c instance.

**Comment by VaslD** [comment]
Why though? Aria 2 is a download manager, not a proxy, VPN, not even a front end for itself.

This is a feature request better suited for whatever front end (control panel, remote control software, GUI, etc.) you’re using.

**Comment by bahusoid** [comment]
I want to add download tasks from console but process them by remote aria2c. I didn't find any console front ends.

Also some front-ends work with aria2c itself not via RPC (for instance [uGet](https://ugetdm.com))  - so proxy mode can also be useful here.

**Comment by BrutalSimplicity** [comment]
Does this not accomplish what you're after?

https://github.com/mayswind/AriaNg

Specifically the section regarding...

> Multi aria2 RPC host support

**Comment by mustaqimM** [comment]
I too wish for functionality similar to https://github.com/nzbget/nzbget from the commandline:

```
$ nzbget -
-                    -- for files (F) and groups (G)-
--append         -A  -- (<options>) <nzb-file/url> Send file/url to server's
  <options> are (multiple options must be separated with space):
       T                    Add file to the top (beginning) of queue
       P                    Pause added files
       C <name>             Assign category to nzb-file
       N <name>             Use this name as nzb-filename
       I <priority>         Set priority (signed integer)

--connect        -C  -- Attach client to server
--edit           -E  -- (F|FN|FR|G|GN|GR|O|H) <action> <IDs/Names/RegExs> Edit items
--list           -L  -- (F|FR|G|GR|O|H|S) (RegEx) Request list of items from server
--log            -G  -- Request last <lines> lines from server's screen-log
--option         -o  -- <name=value> Set or override option in configuration-file
--pause          -P  -- (D|O|S)     Pause server
--printconfig    -p  -- Print configuration and exit
--quit           -Q  -- Shutdown server
--rate           -R  -- Set download rate on server, in KB/s
--reload         -O  -- Reload config and restart all services
--scan           -S  -- (W)         Scan incoming nzb-directory on server
--server         -s  -- Start nzbget as a server in console-mode
--unpause        -U  -- Unpause server
--version        -v  -- Print version and exit
--write          -W  -- <D|I|W|E|G> "Text" Send text to server's log
```

nzbget also has builtin client gui that's accessed via a local url that's really great at keeping track of and controlling everything.

**Comment by VaslD** [comment]
I’m still a bit confused.

So to put in layman’s terms, you guys want to run `aria2c some-command` on your local computer but actually want the command to be processed on a remote host, but not via SSH (which is The Way to run command on a remote host), instead over Aria2’s built-in RPC. In other words basically you want a Web UI without UI; a Web CLI or something like that.

**Comment by zsugabubus** [comment]
> I want to add download tasks from console but process them by remote aria2c. I didn't find any console front ends.

Have you tried [aria2p](https://github.com/pawamoy/aria2p)? From the readme it says it provides a TUI and a CLI, too.

**Comment by tribals** [comment]
To me, I think people want a kinda thin-wrapper CLI which accepts all the range of command line arguments but delegates actual downloading to RPC-enabled aria2c daemon. This makes sense - even when I want to download from the command line, most likely I already set up aria2c daemon. The point here is that a potential thin-wrapper can be used as drop-in replacement for aria2c but downloading will be handled by running daemon.
